e2e/cephfs.go

```diff
@@ -848,6 +848,211 @@ var _ = Describe(cephfsType, func() {
+				if err != nil {
+					e2elog.Failf("failed to list subvolume snapshots: %v", err)
+				}
+				if len(snaps) == 0 {
+					e2elog.Failf("cephFS snapshots list is empty %s/%s", fileSystemName, subvolumes[0].Name)
+				}
+				content, err := getVolumeSnapshotContent(snap.Namespace, snap.Name)
+				if err != nil {
+					e2elog.Failf("failed to get snapshotcontent for %s in namespace %s: %v",
+						snap.Name, snap.Namespace, err)
+				}
+				metadata, err := listCephFSSnapshotMetadata(f,
+					fileSystemName, subvolumes[0].Name, snaps[0].Name, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolume snapshots metadata: %v", err)
+				}
+				if metadata.VolSnapNameKey != snap.Name {
+					e2elog.Failf("failed, snapname expected:%s got:%s",
+						snap.Name, metadata.VolSnapNameKey)
+				} else if metadata.VolSnapNamespaceKey != snap.Namespace {
+					e2elog.Failf("failed, snapnamespace expected:%s got:%s",
+						snap.Namespace, metadata.VolSnapNamespaceKey)
+				} else if metadata.VolSnapContentNameKey != content.Name {
+					e2elog.Failf("failed, contentname expected:%s got:%s",
+						content.Name, metadata.VolSnapContentNameKey)
+				}
+				// Delete the parent pvc before restoring
+				// another one from snapshot.
+				err = deletePVCAndValidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to delete PVC: %v", err)
+				}
+
+				subvols, err := listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolumes: %v", err)
+				}
+				if len(subvols) == 0 {
+					e2elog.Failf("cephFS subvolumes list is empty %s", fileSystemName)
+				}
+				parentSubVol := subvols[0].Name
+
+				// Test Restore snapshot
+				pvcClone, err := loadPVC(pvcClonePath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC: %v", err)
+				}
+				pvcClone.Namespace = f.UniqueName
+				pvcClone.Spec.DataSource.Name = snap.Name
+				// create PVC from the snapshot
+				err = createPVCAndvalidatePV(f.ClientSet, pvcClone, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create pvc clone: %v", err)
+				}
+
+				pvcCloneObj, err := getPersistentVolumeClaim(c, pvcClone.Namespace, pvcClone.Name)
+				if err != nil {
+					e2elog.Logf("error getting pvc %q in namespace %q: %v", pvcClone.Name, pvcClone.Namespace, err)
+				}
+				if pvcCloneObj.Spec.VolumeName == "" {
+					e2elog.Logf("pv clone name is empty %q in namespace %q: %v", pvcClone.Name, pvcClone.Namespace, err)
+				}
+				subvols, err = listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolumes: %v", err)
+				}
+				if len(subvols) == 0 {
+					e2elog.Failf("cephFS subvolumes list is empty %s", fileSystemName)
+				}
+				var cloneSubVol string
+				for _, subvol := range subvols {
+					if subvol.Name == parentSubVol {
+						continue
+					}
+					cloneSubVol = subvol.Name
+
+					break
+				}
```

Comment:
clone subvolume name already exists on the PV object just get the PV object and use it?